### ISSUE TITLE
[Mediaset play] Url apparently changed, can't download from it.

### ISSUE THREAD

**Issue by brsk93** [yt-dlp/yt-dlp#4125]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.05.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Italy

### Description

It looks like Mediaset has changed their URL and now redirects from mediasetplay.mediaset.it to mediasetinfinity.mediaset.it.
I was able to download just fine until a week or so ago, but when I attempt now it doesn't work. It looks like yt-dlp can't handle the new URL and defaults to a generic extractor which also fails.
The video I attempted to download is shown in the verbose log, and plays just fine in browser without needing any account. This is why I have not ticked the last entry in your checklist.

### Verbose log

```shell
yt-dlp --ignore-config -vU https://mediasetinfinity.mediaset.it/video/braveandbeautiful/episodio-113_F310948005000402
[debug] Command-line config: ['--ignore-config', '-vU', 'https://mediasetinfinity.mediaset.it/video/braveandbeautiful/episodio-113_F310948005000402']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8 (No ANSI), error utf-8 (No ANSI), screen utf-8 (No ANSI)
[debug] yt-dlp version 2022.05.18 [b14d523] (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] exe versions: ffmpeg git-2022-06-01-e9107d16 (setts), ffprobe 5.0.1-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3

[debug] Proxy map: {}
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[debug] [generic] Extracting URL: https://mediasetinfinity.mediaset.it/video/braveandbeautiful/episodio-113_F310948005000402
[generic] episodio-113_F310948005000402: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] episodio-113_F310948005000402: Downloading webpage
[generic] episodio-113_F310948005000402: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://mediasetinfinity.mediaset.it/video/braveandbeautiful/episodio-113_F310948005000402
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1395, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1465, in __extract_info
  File "yt_dlp\extractor\common.py", line 642, in extract
  File "yt_dlp\extractor\generic.py", line 4031, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://mediasetinfinity.mediaset.it/video/braveandbeautiful/episodio-113_F310948005000402
```

**Comment by flashdagger** [comment]
Have you tried the id-specific URL format?
`yt-dlp mediaset:F310948005000402`

**Comment by brsk93** [comment]
> Have you tried the id-specific URL format? `yt-dlp mediaset:F310948005000402`

I didn't even know such a thing existed.
Yeah, it works. I'll be using that for now.

**Comment by gamer191** [comment]
What happens if you manually edit the URL back to mediasetplay?

**Comment by gamer191** [comment]
Also, please post a verbose log of ` yt-dlp mediaset:F310948005000402`

**Comment by pukkandan** [comment]
Fixed by 8102a5991babd17707b6b6d39be942382c719d8c

**Comment by Heros99** [comment]
debug: None of the available releases match the specified AssetType, ProtectionScheme, and/or Format preferences.
is it possible to add it to KnownDRMIE?
Thank you very much
